processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -114,3 +137,53 @@ func (e eksDetectorUtils) getConfigMap(ctx context.Context, namespace string, na
 	}
 	return cm.Data, nil
 }
+
+func (e eksDetectorUtils) getClusterName(ctx context.Context) (string, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		return "", err
+	}
+
+	ec2Svc := ec2metadata.New(sess)
+	region, err := ec2Svc.Region()
+	if err != nil {
+		return "", err
+	}
+
+	svc := ec2.New(sess, aws.NewConfig().WithRegion(region))
+	instanceIdentityDocument, err := ec2Svc.GetInstanceIdentityDocumentWithContext(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	instances, err := svc.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: []*string{
+			aws.String(instanceIdentityDocument.InstanceID),
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	clusterName := e.getClusterNameTagFromReservations(instances.Reservations)
+	if len(clusterName) == 0 {
+		return clusterName, fmt.Errorf("Failed to detect EKS cluster name. No tag for cluster name found on EC2 instance")
+	}
+	return clusterName, nil
+}
+
+func (e eksDetectorUtils) getClusterNameTagFromReservations(reservations []*ec2.Reservation) string {
+	for _, reservation := range reservations {
+		for _, instance := range reservation.Instances {
+			for _, tag := range instance.Tags {
+				if *tag.Key == clusterNameAwsEksTag || *tag.Key == clusterNameEksTag {
```

Comment:
Is the key guaranteed to be not `nil`?

Reply:
I just checked, there's no enforcement in place to guarantee it's non-nil. I'll add a check here. 👍